src/gicv3.rs

```diff
@@ -16,10 +16,17 @@ use core::{
     hint::spin_loop,
     mem::size_of,
 };
+use thiserror_no_std::Error;
 
 /// The offset in bytes from `RD_base` to `SGI_base`.
 const SGI_OFFSET: usize = 0x10000;
 
+#[derive(Error, Debug)]
```

Comment:
Derive `Clone`, `Copy`, `Eq` and `PartialEq` too, it will make life easier for users of the crate, especially for tests.